### ISSUE TITLE
Command line arguments are broken!

### ISSUE THREAD

**Issue by pazz** [pazz/alot#942]
I found that the `-n` switch seems problematic:

```
alot -n idontexist
Could not read idontexist and/or /home/pazz/.local/lib/python2.7/site-packages/alot-0.4.dev0-py2.7.egg/alot/settings/../defaults/notmuch.rc.spec
```

Also, the `-d` switch for the debug log level fails
```
alot -d debug
Traceback (most recent call last):
  File "/home/pazz/.local/bin/alot", line 11, in <module>
    load_entry_point('alot==0.4.dev0', 'console_scripts', 'alot')()
  File "/home/pazz/.local/lib/python2.7/site-packages/alot-0.4.dev0-py2.7.egg/alot/__main__.py", line 116, in main
    args.parseOptions()  # When given no argument, parses sys.argv[1:]
  File "/usr/lib/python2.7/dist-packages/twisted/python/usage.py", line 253, in parseOptions
    self._dispatch[optMangled].dispatch(optMangled, arg)
  File "/usr/lib/python2.7/dist-packages/twisted/python/usage.py", line 59, in dispatch
    value = self.coerce(value)
TypeError: debuglogstring() takes exactly 2 arguments (1 given)
```

**Comment by lucc** [comment]
I did a git bisect and it seems to be 581ed2987bd456d2894637a30bb5a14a4caa5f9b that introduced this.

I did 
```sh
git bisect start master 0.4
git bisect run sh -c 'PYTHONPATH=. python2 -m alot -l out -d debug'
```

**Comment by dcbaker** [comment]
I think just reverting my patch is the easiest solution. The changes were meant to be cosmetic only, if we want to fix them again we can revist them later.

**Comment by pazz** [comment]
this was quick, thanks.
I wouldn't mind replacing the `twisted.usage` option parsing by argparse. The only reason why I originally didn't is that at that point (still?) argparse wasn't able to model subcommands (`alot search --help`..)

How would we test this?

**Comment by dcbaker** [comment]
It's a bit of a functional test, but probably with `subprocess.check_call`.

argparse does handle subcommands, but there are some corners. If you're interested in moving to argparse I'd be willing to do that work. I've done a lot of work with argparse.

**Comment by pazz** [comment]
@dcbaker sure, I'd be interested. Alot already uses argparse for commands internally, and I'm generally not a huge fan of twisted anyway. It's not a priority of course.

**Comment by lucc** [comment]
@dcbaker If I rightly remember some old comment argparse did not support optional subcommands. http://bugs.python.org/issue9253 seems to say the same.

But there is a simple hack which I applied to khard, see https://github.com/scheibler/khard/blob/master/khard/khard.py#L1170 and https://github.com/scheibler/khard/blob/master/khard/khard.py#L1455 

Maybe that helps.

**Comment by dcbaker** [comment]
@lucc that's fixed in python 3.3 thankfully. My workaround has been to just add an "invisible" default option that I inject if one of the other options isn't present. your workaround looks a bit more elegent than mine.

**Comment by lucc** [comment]
I now think that it will be even easier for us as we do not read a configurable default command from the config file. I hope you did not start working on it because I'm very motivated to try this now :)
Stay tuned.
